core/lib/types/src/commitment.rs

```diff
@@ -577,181 +573,134 @@ impl L1BatchCommitment {
             commitment,
         }
     }
-}
-
-#[cfg(test)]
-mod tests {
-    use serde::{Deserialize, Serialize};
-    use serde_with::serde_as;
-
-    use crate::{
-        commitment::{
-            L1BatchAuxiliaryOutput, L1BatchCommitment, L1BatchMetaParameters,
-            L1BatchPassThroughData,
-        },
-        l2_to_l1_log::{L2ToL1Log, UserL2ToL1Log},
-        writes::{InitialStorageWrite, RepeatedStorageWrite},
-        ProtocolVersionId, H256, U256,
-    };
 
-    #[serde_as]
-    #[derive(Debug, Serialize, Deserialize)]
-    struct ExpectedOutput {
-        #[serde_as(as = "serde_with::hex::Hex")]
-        l2_l1_bytes: Vec<u8>,
-        l2_l1_linear_hash: H256,
-        l2_l1_root_hash: H256,
-        #[serde_as(as = "serde_with::hex::Hex")]
-        initial_writes_bytes: Vec<u8>,
-        initial_writes_hash: H256,
-        #[serde_as(as = "serde_with::hex::Hex")]
-        repeated_writes_bytes: Vec<u8>,
-        repeated_writes_hash: H256,
-        #[serde_as(as = "serde_with::hex::Hex")]
-        pass_through_bytes: Vec<u8>,
-        pass_through_hash: H256,
-        #[serde_as(as = "serde_with::hex::Hex")]
-        meta_params_bytes: Vec<u8>,
-        meta_params_hash: H256,
-        #[serde_as(as = "serde_with::hex::Hex")]
-        auxiliary_bytes: Vec<u8>,
-        auxiliary_hash: H256,
-        commitment_hash: H256,
+    pub fn artifacts(&self) -> L1BatchCommitmentArtifacts {
```

Comment:
Dumb question: Can't `L1BatchCommitment` be used in `save_l1_batch_commitment_artifacts()` directly (IIUC, that's the only non-test use of `L1BatchCommitmentArtifacts`)? Would it break an abstraction boundary?